src/docs/arch-session.dox

```diff
@@ -0,0 +1,53 @@
+/*! @arch_page arch-session Session
+
+@section arch_session_def Definition
+After a @ref arch-connection has been established between the application and WiredTiger, the
+application can start sending requests to WiredTiger using a session. A session is internally
+represented by WT_SESSION and plays an important role since almost all operations require one.
+
+A session can only be created through an existing connection with the API
+WT_CONNECTION::open_session. It is possible to create multiple sessions through the same connection.
+In fact, one connection can have multiple sessions but one session can only be associated to one
+connection. The maximum number of sessions is set through the configuration item \c session_max as
+part of the configuration string given in ::wiredtiger_open.
+
+Sessions that are created by the calling application are called "user session". It exists another
+type of session called "internal session". Internal sessions are created and managed by WiredTiger
+internally and they are associated to global services at runtime such as @ref arch-eviction, @ref
+arch-checkpoint, @ref arch-rts, ...
+
+@section arch_session_ops Operations
+The different operations a session can request to WiredTiger are related to cursors handles, table
+operations and transactions. You can read the complete description of each possible operation in the
+documentation related to WT_SESSION.
+
+@section arch_session_txn Transactions
+It is possible to group a number of operations within a session, in other words, multiple operations
+can be treated as a single atomic operation. This can be done using @ref arch-transaction. When
+multiple operations are part of a transaction, they must either all be successful or fail;
+WiredTiger will abort any successful operation before a failure within the same transaction.
+Finally, one session can hold a unique transaction at any given time and this transaction only
+belongs to that same session.
+
+@section arch_session_cur Cursors
+Session can perform operations on multiple collections using cursors (see @ref arch-cursor for more
+details). Indeed, a session can have multiple cursors of any types and they all share the same
```

Comment:
`any type`

Reply:
Fixed in 4d91ebde292489e965f32a661fcb02d6f65084dc